public/app/features/alerting/unified/components/rules/RuleActionsButtons.tsx

```diff
@@ -0,0 +1,211 @@
+          icon="eye"
+          href={createViewLink(rulesSource, rule, returnTo)}
+        >
+          <DontShowIfSmallDevice>View</DontShowIfSmallDevice>
+        </LinkButton>
+      </Tooltip>
+    );
+  }
+
+  if (isEditable && rulerRule && !isFederated && !isProvisioned) {
+    const sourceName = getRulesSourceName(rulesSource);
+    const identifier = ruleId.fromRulerRule(sourceName, namespace.name, group.name, rulerRule);
+
+    const editURL = urlUtil.renderUrl(
+      `${config.appSubUrl}/alerting/${encodeURIComponent(ruleId.stringifyIdentifier(identifier))}/edit`,
+      {
+        returnTo,
+      }
+    );
+
+    if (isViewMode) {
+      buttons.push(
+        <ClipboardButton
+          key="copy"
+          icon="copy"
+          onClipboardError={(copiedText) => {
+            notifyApp.error('Error while copying URL', copiedText);
+          }}
+          className={style.button}
+          size="sm"
+          getText={buildShareUrl}
+        >
+          Copy link to rule
+        </ClipboardButton>
+      );
+    }
+
+    buttons.push(
+      <Tooltip placement="top" content={'Edit'}>
+        <LinkButton className={style.button} size="xs" key="edit" variant="secondary" icon="pen" href={editURL}>
+          <DontShowIfSmallDevice>Edit</DontShowIfSmallDevice>
+        </LinkButton>
+      </Tooltip>
+    );
+  }
+
+  if (isRemovable && rulerRule && !isFederated && !isProvisioned) {
+    buttons.push(
+      <Tooltip placement="top" content={'Delete'}>
+        <Button
+          className={style.button}
+          size="xs"
+          type="button"
+          key="delete"
+          variant="secondary"
+          icon="trash-alt"
+          onClick={() => setRuleToDelete(rule)}
+        >
+          <DontShowIfSmallDevice>Delete</DontShowIfSmallDevice>
+        </Button>
+      </Tooltip>
+    );
+  }
+
+  if (buttons.length) {
+    return (
+      <>
+        <div className={style.wrapper}>
+          <HorizontalGroup width="auto">
+            {buttons.length ? buttons.map((button, index) => <div key={index}>{button}</div>) : <div />}
+          </HorizontalGroup>
+        </div>
+        {!!ruleToDelete && (
+          <ConfirmModal
+            isOpen={true}
+            title="Delete rule"
+            body="Deleting this rule will permanently remove it from your alert rule list. Are you sure you want to delete this rule?"
+            confirmText="Yes, delete"
+            icon="exclamation-triangle"
+            onConfirm={deleteRule}
+            onDismiss={() => setRuleToDelete(undefined)}
+          />
+        )}
+      </>
+    );
+  }
+
+  return null;
+};
+
+function inViewMode(pathname: string): boolean {
+  return pathname.endsWith('/view');
+}
+
+export const getStyles = (theme: GrafanaTheme2) => ({
+  wrapper: css`
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+    flex-wrap: wrap;
+  `,
+  button: css`
+    height: 24px;
+    font-size: ${theme.typography.size.sm};
+  `,
```

Comment:
I noticed the icons are not centered when the text of the button is hidden:

![image](https://user-images.githubusercontent.com/6271380/198393576-94fb84c8-2512-41e5-a0da-c1bd806f30b0.png)

This happens because the icon's `svg` has a `margin-right` of `8px`. One way to fix it could be to remove this margin from the icon and apply a `margin-left: 8px` to the text instead:

```
  button: css`
    height: 24px;
    font-size: ${theme.typography.size.sm};

    svg {
      margin-right: 0;
    }
  `,

  buttonText: css`
    margin-left: 8px;
  `,
  ```
  
  Note that I added the `buttonText` class to the text in the `DontShowIfSmallDevice` component as
  
  ```
  if (isSmallScreen) {
    return null;
  } else {
    return <div className={style.buttonText}>{children}</div>;
  }
  ```
  
  
  With these changes it centers correctly:
  
  
![2022-10-27 17 45 43](https://user-images.githubusercontent.com/6271380/198394511-af2f5bf5-3c69-462d-982e-00f1e7aca9a4.gif)

Reply:
nice! thank you @VikaCep for your detailed explanation 🚀

Reply:
fixed!🙌